core_contracts/governance/governance.py

```diff
@@ -559,6 +592,8 @@ def addNewDataSource(self, _data_source_name: str, _contract_address: Address) -
         Add a new data source to receive BALN tokens. Starts with a default of
         zero percentage of the distribution.
         """
+        if type(_contract_address) != Address:
```

Comment:
This may get rejected in audit. They don't like parameter types to be uncertain. Please ask about this before we submit so we can correct it if needed without losing time.

Reply:
Shall i remove this in this PR ?